File: Modules/ConfigModule/ConfigModule/Classes/ModuleConfigServiceImpl.m
//
//  ModuleConfigServiceImpl.m
//  ConfigModule
//
//  Created by NeroXie on 2023/2/6.
//

#import "ModuleConfigServiceImpl.h"

@interface ModuleConfigServiceImpl()

@property (nonatomic, copy) NSString *appScheme;

@property (nonatomic, strong) Class tabBarControllerType;

@property (nonatomic, copy) NSArray<NSString *> *tabBarItems;


@end

@implementation ModuleConfigServiceImpl

- (instancetype)init {
    if (self = [super init]) {
        // 模拟从配置文件中读取
        
        self.appScheme = @"nero";
        self.tabBarControllerType = NSClassFromString(@"TabBarController.TabBarController");
        self.tabBarItems = @[@"example", @"house", @"user"];
    }
    
    return self;
}

@end

File: Modules/ApplicationModule/ApplicationModule/Classes/ModuleConfigServiceImpl.swift
//
//  ModuleConfigServiceImpl.swift
//  ApplicationModule
//
//  Created by NeroXie on 2022/11/20.
//

import Foundation
import NNModule_swift
import ModuleServices

class ModuleConfigServiceImpl: NSObject, ModuleConfigService {
    
    private(set) var appScheme: String = ""
    
    private(set) var tabBarControllerType: UITabBarController.Type = UITabBarController.self
    
    private(set) var tabBarItems = [String]()
    
    required override init() {
        super.init()
        
        // 从主工程配置文件中读取
        appScheme = "nero"
        tabBarItems = ["example", "house", "user"]
        if let cls = NSClassFromString("TabBarController.TabBarController"), let tabBarType = cls as? UITabBarController.Type {
            tabBarControllerType = tabBarType
        }
    }
}
